addons/project_purchase/controllers/catalog.py

```diff
@@ -0,0 +1,22 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.http import request, route
+from odoo.addons.product.controllers.catalog import ProductCatalogController
+
+
+class ProjPurchaseCatalogController(ProductCatalogController):
```

Comment:
Don't be afraid to right a long class name. 😄 

```suggestion
class ProjectPurchaseCatalogController(ProductCatalogController):
```